From 77b3fb21b644c230ac8e7ba96f7e13ae3eed3504 Mon Sep 17 00:00:00 2001
From: Jason HU <yongqi.hu@windriver.com>
Date: Fri, 25 Jun 2010 11:13:56 +0800
Subject: [PATCH] sysctl: sysctl table check failure fix for mercury

The check failure for /net/mercury was fixed, which will be
reported when CONFIG_SYSCTL_SYSCALL_CHECK is turned on.

Signed-off-by: Jason HU <yongqi.hu@windriver.com>
---
 kernel/sysctl_check.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/kernel/sysctl_check.c b/kernel/sysctl_check.c
index c35da23..edbdf67 100644
--- a/kernel/sysctl_check.c
+++ b/kernel/sysctl_check.c
@@ -690,6 +690,13 @@ static const struct trans_ctl_table trans_net_irda_table[] = {
 	{}
 };
 
+#ifdef CONFIG_MERCURY
+static const struct trans_ctl_table trans_net_mercury_table[] = {
+	{ NET_MERC_MAX_DGRAM_QLEN,	"max_dgram_qlen" },
+	{}
+};
+#endif
+
 static const struct trans_ctl_table trans_net_table[] = {
 	{ NET_CORE,		"core",		trans_net_core_table },
 	/* NET_ETHER not used */
@@ -713,6 +720,9 @@ static const struct trans_ctl_table trans_net_table[] = {
 	{ NET_DCCP,		"dccp",		trans_net_dccp_table },
 	{ NET_IRDA,		"irda",		trans_net_irda_table },
 	{ 2089,			"nf_conntrack_max" },
+#ifdef CONFIG_MERCURY
+	{ NET_MERCURY,		"mercury",	trans_net_mercury_table },
+#endif
 	{}
 };
 
-- 
1.7.0.4

